### ISSUE TITLE
Sonoff & BME280

### ISSUE THREAD

**Issue by echo4019** [arendst/Sonoff-Tasmota#1865]
Hi,

Can anyone describe the process of connecting a BME280 sensor to a sonoff basic? 

Any help is appreciated.

**Comment by curzon01** [comment]
1. based on [Sonoff-Basic GPIO Locations](https://github.com/arendst/Sonoff-Tasmota/wiki/Sonoff-Basic#gpio-locations) connect 
```
BME280-3.3V -> Sonoff-3.3V
BME280-GND -> Sonoff-GND
BME280-SCL -> Sonoff-TX
BME280-SDA -> Sonoff-RX
```

2. On Tasmota web interface go to "Configration->Configure Module" and set
```
GPIO1 Serial Out -> I2C SCL
GPIO3 Serial In -> I2C SDA
```
Restart

**Comment by echo4019** [comment]
When checking the Sonoff-Basic GPIO Locations it mentions to access GPIO 04 you must solder wire to pin on ESP chip. Is this referring to older versions of the sonoff basic?

Also on the tasmota web interface under configure module I have no option on the drop down list for I2C SCL or I2C SDA. 

Thanks

**Comment by echo4019** [comment]
My bad I had disabled #define USE_I2C on the config file.

**Comment by echo4019** [comment]
- platform: mqtt
    name: "BME280 Temperature"
    state_topic: "tele/bme280/SENSOR"
    unit_of_measurement: "°C"
    value_template: "{{ value_json.Temperature | round(1) }}"

Is this ok for the config.yaml file for home assistant?

**Comment by curzon01** [comment]
In the configuration above I don't use GPIO4

**Comment by jumblies** [comment]
Were you guys able to do this on a sonoff basic?  I'm struggling with this one.  Did you use a pullup on VCC and Data?

**Comment by curzon01** [comment]
Comment above contains description for the 4 wires you need, link inside contains pictures of the sonoff basic. If you use a BME which comes soldered on one of common breakout boards you don't need any pulls, only connect the 4 wires and configure the Tasmota as described.

**Comment by glamke** [comment]
Solved.  It was a circuitous route but it worked perfect once I managed to upgrade to the latest binary.  I had a very difficult time getting flashes done properly with the binaries.  Flashes worked, but then units wouldn't respond.  Another issue for another thread.  

Thank you for your help

**Comment by curzon01** [comment]
So can you pls close your issue?

**Comment by jumblies** [comment]
This closes issue #1865 for me, but I didn't open it
